### PR TITLE
Integrating with pvc-transfer library

### PR DESCRIPTION
**Pull request by pranavgaikwad**
- Removes `konveyor/crane-lib` dependency
- Updates `transfer-pvc` subcommand to use `backube/pvc-transfer` library instead of `crane-lib`
- Refactors struct for command flags:
  - Seggregates flags for endpoint and pvc
  - Introduces a new flag for ingress-class
  - Updates endpoint type flag to structured

### CODE REVIEW

File: cmd/transfer-pvc/transfer-pvc.go [hunk 1]
```diff
@@ -355,11 +450,23 @@ func getNodeNameForPVC(srcClient client.Client, namespace string, pvcName string
...
+	}
+	return string(password)
+}
+
+func garbageCollect(srcClient client.Client, destClient client.Client, labels map[string]string, endpoint endpointType, namespace mappedNameVar) error {
 	srcGVK := []client.Object{
 		&corev1.Pod{},
 		&corev1.ConfigMap{},
 		&corev1.Secret{},
+		&corev1.ServiceAccount{},
```
Comment (alaypatel07) [inline cmd/transfer-pvc/transfer-pvc.go:469]: is this list pending some more resources for garbage collection? I could be wrong but I recall more RBAC resources that were added in the lib needing garbage collection

**Review by alaypatel07 — COMMENTED** [review]
Added some nits, overall this looks great

File: cmd/transfer-pvc/transfer-pvc.go [hunk 2]
```diff
@@ -186,151 +254,177 @@ func (t *TransferPVCOptions) getClientFromContext(ctx string) (client.Client, er
...
-	switch t.Endpoint {
-	case endpointRoute:
-		e = createAndWaitForRoute(destPVC, destClient)
-	case endpointNginx:
-		e = createAndWaitForIngress(destPVC, destClient)
-	default:
-		log.Fatalf("unsupported endpoint type %s\n", t.Endpoint)
+	waitForEndpoint(e, destClient)
+
+	if _, err := e.IsHealthy(context.TODO(), destClient); err != nil {
```
Comment (alaypatel07) [inline cmd/transfer-pvc/transfer-pvc.go:319]: nit: line 317 not needed anymore if above returns error

**Review by alaypatel07 — COMMENTED** [review]
overall LGTM, will leave the tags for @djzager

File: cmd/transfer-pvc/transfer-pvc.go (hunk 2 above)
Comment (pranavgaikwad) [reply cmd/transfer-pvc/transfer-pvc.go:319]: @alaypatel07 its still needed actually. The wait function polls. so maybe 319 is not needed

Comment (alaypatel07) [reply cmd/transfer-pvc/transfer-pvc.go:319]: sorry I meant "L319 not needed anymore if L317 returns error", +1

**Review by djzager — APPROVED** [review]
/lgtm

File: cmd/transfer-pvc/transfer-pvc.go (hunk 2 above)
Comment (pranavgaikwad) [reply cmd/transfer-pvc/transfer-pvc.go]: @alaypatel07 updated

**Review by djzager — APPROVED** [review]
/lgtm

**Pull request merged**